plugin-build/src/main/kotlin/io/sentry/android/gradle/instrumentation/androidx/room/visitor/RoomCallMethodVisitor.kt

```diff
@@ -0,0 +1,130 @@
+package io.sentry.android.gradle.instrumentation.androidx.room.visitor
+
+import io.sentry.android.gradle.instrumentation.AbstractSpanAddingMethodVisitor
+import io.sentry.android.gradle.instrumentation.util.ReturnType
+import org.objectweb.asm.Label
+import org.objectweb.asm.MethodVisitor
+import org.objectweb.asm.Opcodes
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.atomic.AtomicReference
+
+class RoomCallMethodVisitor(
+    private val returnType: ReturnType,
+    api: Int,
+    methodVisitor: MethodVisitor,
+    descriptor: String?
+) : AbstractSpanAddingMethodVisitor(
+    api = api,
+    methodVisitor = methodVisitor,
+    descriptor = descriptor
+) {
+
+    private val label5 = Label()
+    private val label6 = Label()
+    private val label7 = Label()
+    private val label8 = Label()
+    private val label9 = Label()
```

Comment:
mm, I think you mean that I extracted them into the abstract class rather, but we still need to use labels. This visitor is just different from those others, since we need to inject a catch-block into the existing try-finally, hence more labels and jumps

Reply:
yeah could be, thanks